sklearn/tree/tree.py

```diff
@@ -510,6 +515,110 @@ def decision_path(self, X, check_input=True):
         X = self._validate_X_predict(X, check_input)
         return self.tree_.decision_path(X)
 
+    def prune_tree(self):
```

Comment:
I think this needs to be called after fit automatically to facilitate cross validation etc.

I wonder if this should instead be a public function?